uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-leave-absences-my-leave-report.js

```diff
@@ -60,6 +60,63 @@ module.exports = (function () {
           resolve(this.waitForModal('ssp-leave-request', '.chr_leave-request-modal__form'));
         }.bind(this));
       }.bind(this));
+    }*/
+    /**
+     * User clicks on the edit/respond action
+     * @param {Number} row number corresponding to leave request in the list
+     * @return {Promise}
+     */
+    editRequest: function (row) {
+      var casper = this.casper;
+
+      casper.then(function () {
+        casper.click('body > ul.dropdown-menu:nth-of-type(' + (row || 1) + ') li:first-child a');
+        // As there are multiple spinners it takes more time to load up
+        casper.waitWhileVisible('.modal-content .spinner:nth-child(1)');
+        casper.waitWhileVisible('leave-request-popup-details-tab .spinner');
+      });
+
+      return this;
+    },
+
+    openCommentsTab: function () {
+      var casper = this.casper;
+      var selector = '.chr_leave-request-modal__tab .uib-tab:nth-of-type(2) a';
+
+      casper.waitForSelector(selector)
+        .then(function () {
+          casper.click(selector);
```

Comment:
I think you don't need the .then. Can you try this and see if it works:

```js
casper.waitForSelector(selector);
casper.click(selector);
```

In case it doesn't work, use `.bind` instead of doing `var casper = this.casper`

Reply:
Done.